great_expectations/data_context/data_context/context_factory.py

```diff
@@ -140,6 +141,57 @@ def get_validator(self, batch_request: BatchRequest) -> Validator:
             )
         return self._project.get_validator(batch_request=batch_request)
 
+    def is_using_cloud(self) -> bool:
+        if not self._project:
```

Comment:
To DRY this up, can you make this if check a private method and call it here and elsewhere?